### PR TITLE
run literature in two steps, processing and ML

### PR DESCRIPTION
**Pull request by mkarmona**

### CODE REVIEW

**Review by JarrodBaker — COMMENTED** [review]
We're starting to collect these scripts in the ETL repo too. Do you think it's time for us to start a new repo with all the Dataproc scripts and generalise some of the logic into common parameterised scripts (start cluster, start job, etc.)?

**Comment by cmalangone** [comment]
> We're starting to collect these scripts in the ETL repo too. Do you think it's time for us to start a new repo with all the Dataproc scripts and generalise some of the logic into common parameterised scripts (start cluster, start job, etc.)?

@JarrodBaker I agree. We should have a common place where share/update these scripts. What about if we start with some gist?

**Review by cmalangone — APPROVED** [review]
Can you please leave a note about the reason why you use
val prefix = Random.alphanumeric.take(6) with cardinality 6 inside the method disambiguate

File: src/main/scala/io/opentargets/etl/literature/Grounding.scala [hunk 1]
```diff
@@ -538,8 +587,18 @@ object Grounding extends Serializable with LazyLogging {
...
-    val mappedLabels = mapEntities(sentences, luts, pipeline, pipelineColumns).cache()
-    val resolvedEntities = resolveEntities(sentences, mappedLabels)
+    val mappedLabels =
+      mapEntities(sentences, luts, pipeline, pipelineColumns)
+
+    logger.info("producing grounding dataset")
+    Helpers.writeTo(
+      Map("mappedLabels" -> IOResource(mappedLabels, empcConfiguration.outputs.grounding)))
+
+    val aux = Helpers.readFrom(Map("mappedLabels" -> empcConfiguration.outputs.grounding))
```
Comment (JarrodBaker) [inline src/main/scala/io/opentargets/etl/literature/Grounding.scala]: Why are you writing out in line 594 and then reading it straight back in again?

File: src/main/scala/io/opentargets/etl/literature/Grounding.scala [hunk 2]
```diff
@@ -152,27 +192,36 @@ object Grounding extends Serializable with LazyLogging {
...
+
+    logger.info("ground and take rank 1 from the mapped ones")
     val w = Window.partitionBy($"type", $"labelN").orderBy(scoreC.desc)
     val mappedLabel = labels
-      .select("type", "label", "labelN")
       .join(luts, Seq("type", "labelN"), "left_outer")
       .withColumn("isMapped", $"keywordId".isNotNull)
       .filter($"isMapped" === true)
+      .select(selelectedCols.map(col): _*)
+      .repartition(2048)
```
Comment (JarrodBaker) [inline src/main/scala/io/opentargets/etl/literature/Grounding.scala]: Any reason for '2048'? Performance tuning?

**Review by JarrodBaker — APPROVED** [review]

File: src/main/scala/io/opentargets/etl/literature/Grounding.scala (hunk 2 above)
Comment (mkarmona) [reply src/main/scala/io/opentargets/etl/literature/Grounding.scala]: @JarrodBaker I am tryuing to improve the segments to process but I realised that is not the place to put it as it writes what it wants.

**Pull request merged**